common/Rectangle.hpp

```diff
@@ -63,39 +85,49 @@ struct Rectangle
             _y2 = rectangle._y2;
     }
 
-    void setLeft(int x1) { _x1 = x1; }
+    constexpr void setLeft(int x1) /*noexcept*/ { _x1 = x1; }
 
-    int getLeft() const { return _x1; }
+    constexpr int getLeft() const /*noexcept*/ { return _x1; }
 
-    void setRight(int x2) { _x2 = x2; }
+    constexpr void setRight(int x2) /*noexcept*/ { _x2 = x2; }
 
-    int getRight() const { return _x2; }
+    constexpr int getRight() const /*noexcept*/ { return _x2; }
 
-    void setTop(int y1) { _y1 = y1; }
+    constexpr void setTop(int y1) /*noexcept*/ { _y1 = y1; }
 
-    int getTop() const { return _y1; }
+    constexpr int getTop() const /*noexcept*/ { return _y1; }
 
-    void setBottom(int y2) { _y2 = y2; }
+    constexpr void setBottom(int y2) /*noexcept*/ { _y2 = y2; }
 
-    int getBottom() const { return _y2; }
+    constexpr int getBottom() const /*noexcept*/ { return _y2; }
 
-    int getWidth() const { return _x2 - _x1; }
+    constexpr int getWidth() const /*noexcept*/ { return _x2 - _x1; }
 
-    int getHeight() const { return _y2 - _y1; }
+    constexpr int getHeight() const /*noexcept*/ { return _y2 - _y1; }
 
-    bool isValid() const { return _x1 <= _x2 && _y1 <= _y2; }
+    constexpr bool isValid() const /*noexcept*/ { return _x1 <= _x2 && _y1 <= _y2; }
 
-    bool hasSurface() const { return _x1 < _x2 && _y1 < _y2; }
+    constexpr bool hasSurface() const /*noexcept*/ { return _x1 < _x2 && _y1 < _y2; }
 
-    bool intersects(const Rectangle& rOther)
+    /// Returns whether this Rectangle intersects (partially contains) given Rectangle.
+    constexpr bool intersects(const Rectangle& rOther) const /*noexcept*/
     {
-        Util::Rectangle intersection;
-        intersection._x1 = std::max(_x1, rOther._x1);
-        intersection._y1 = std::max(_y1, rOther._y1);
-        intersection._x2 = std::min(_x2, rOther._x2);
-        intersection._y2 = std::min(_y2, rOther._y2);
-        return intersection.isValid();
+        const int x1 = std::max(_x1, rOther._x1);
```

Comment:
If there are no unit tests for this code; why change it ? If we are changing it in a not totally obvious way - there should be unit tests.

Reply:
WhiteBoxTests.cpp does validate Rectangle intersects after consolidation via TileDesc::rectanglesIntersect